### ISSUE TITLE
The benchmark cache is not invalidated when data changes

### ISSUE THREAD

**Issue by glpuga** [Ekumen-OS/lambkin#95]
### Bug description

After a failed and resumed run, the report generation totally ignored one of the five datasets and generated no data for it.

The issue was due to the benchmark cache having become stale, and not getting updated when on the resumed run the last dataset was added after having been absent when the first run failed.

The issue was fixed by removing the cache file after confirming in the code that it would be regenerated.

### How to reproduce

I guess that anything that fills the cache and then changes data will do

**Expected behavior**

The cache should have some clear invalidation criteria.

The best one for this use case is probably the complete invalidation of the cache contents on each run of the command.

**Actual behavior**

The cache keeps state across runs of shepherd, causing stale data to be used instead of actual state.

### Additional context

.

**Comment by hidmic** [comment]
Indeed. [`DataFrame` caching](https://github.com/Ekumen-OS/lambkin/blob/52da4cf13fff0928ac0e1ef9fef5d4f43832bea9/src/core/lambkin-shepherd/src/lambkin/shepherd/data/pandas.py#L86-L119) is not paying attention to the data source because it doesn't know about it. It should at least pick up what the target iterations are for the underlying function and perform a timestamp check.
